cub/cub/device/dispatch/dispatch_radix_sort.cuh

```diff
@@ -1104,25 +1084,26 @@ struct DispatchRadixSort
       // Small, single tile size
       return InvokeSingleTile(kernel_source.RadixSortSingleTileKernel(), wrapped_policy);
     }
-// We guard this because this will instantiate and compile both branches, which is not necessary if we know whether we
-// are using one sweep at compile-time
-#ifdef CCCL_C_EXPERIMENTAL
-    else if (wrapped_policy.IsOnesweep())
-    {
-      // Regular size
-      return InvokeManyTiles(detail::bool_constant_v<true>, wrapped_policy);
-    }
-    else
+
+    if
+#ifndef CCCL_C_EXPERIMENTAL
+      // Branching at compile time prevents instantiation and compilation of both branches
+      constexpr
+#endif // CCCL_C_EXPERIMENTAL
```

Comment:
Please move a macro to `__cccl/compiler.h` or `__cccl/dialect.h` I really want to avoid those complicated clauses with a ton of constexpr sprinkled around
```suggestion
    if _CCCL_CONSTEXPR_IN_CPP
```
Please find a better name though